clang/include/clang/Tooling/DependencyScanning/DependencyScanningWorker.h

```diff
@@ -151,14 +159,16 @@ class DependencyScanningWorker {
   /// (passed in the constructor).
   llvm::IntrusiveRefCntPtr<DependencyScanningWorkerFilesystem> DepFS;
 
+  friend class CompilerInstanceWithContext;
```

Comment:
I think you can either drop the `class` here or drop the entire forward declaration at the top of this file.